packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -376,18 +384,21 @@ def _json_input_example_template(self, builder: BuilderType) -> List[str]:
 
         if (
             isinstance(body_param.type, (ListType, DictionaryType))
-            and self.code_model.options["models_mode"]
+            and self.code_model.options["models_mode"] == "msrest"
         ):
             return template
 
-        if isinstance(body_param.type, ModelType) and body_param.type.base != "json":
+        if isinstance(body_param.type, ModelType) and body_param.type.base == "msrest":
             return template
 
         json_type = body_param.type
         if isinstance(body_param.type, CombinedType):
-            if body_param.type.json_subtype is None:
+            target_model_type = body_param.type.target_model_subtype(
```

Comment:
thanks, much cleaner! I'm wondering though, don't we always want to generate docs for both `JSONModelType` and `DPGModelType`? I thought this is what we landed on, but correct me if I'm wrong. If so, then we won't need the target type tuple anymore